addons/mail/static/src/models/file_uploader_view/file_uploader_view.js

```diff
@@ -4,10 +4,147 @@ import { registerModel } from '@mail/model/model_core';
 import { attr, many2one, one2one } from '@mail/model/model_field';
 import { clear, replace } from '@mail/model/model_field_command';
 
+import core from 'web.core';
+
+const geAttachmentNextTemporaryId = (function() {
+    let tmpId = 0;
+    return () => {
+        tmpId += 1;
+        return tmpId;
+    };
+})();
+
 registerModel({
     name: 'FileUploaderView',
     identifyingFields: [['activityView', 'attachmentBoxView', 'composerView']],
+    lifecycleHooks: {
+        _created() {
+            this.onChangeAttachment = this.onChangeAttachment.bind(this);
+        }
+    },
     recordMethods: {
+        /**
+         * @param {FileList|Array} files
+         * @returns {Promise}
+         */
+        async uploadFiles(files) {
+            await this._performUpload({ files });
+            this.fileInputRef.el.value = '';
```

Comment:
This needs to be guarded after `await`, both `this.fileInputRef` and `this.fileInputRef.el`, due to async behavior in rendering, and the long delay in upload making this likely to happen.